providers/providers.go

```diff
@@ -78,13 +80,21 @@ func newProviderDataFromConfig(providerConfig options.Provider) (*ProviderData,
 		ClientSecretFile: providerConfig.ClientSecretFile,
 	}
 
+	client, err := util.GetHTTPClient(providerConfig.TLSCertFile, providerConfig.TLSKeyFile, providerConfig.SSLInsecureSkipVerify, providerConfig.UseSystemTrustStore, providerConfig.CAFiles...)
```

Comment:
The whole `newProviderDataFromConfig` method is getting rather long and it reached our code climate limit of no more than 5 returns in a single method. We should probably split it into multiple sections for better clarity.